src/scripts/hooks/utils/Ui5ExtensionMocker.ts

```diff
@@ -0,0 +1,30 @@
+interface MockResponse {
+  payload: string;
+  params: {
+    statusCode: number;
+    fetchResponse: boolean;
+  }
+}
+
+export class Ui5ExtensionMocker {
+  private static readonly urlRegexps: RegExp[] = [
+    /.*\/xRayControls\/.*/gm,
+    /.*joule.*/gm,
+    /.*walkme.*/gm
+  ];
+
+  private static readonly mockResponse: MockResponse = {
+    payload: "File not found",
```

Comment:
Maybe something more transparent, like: 
```suggestion
    payload: "File intentionally mocked by Qmate to improve performance",
```